streams/src/test/java/org/apache/kafka/streams/processor/internals/NamedTopologyTest.java

```diff
@@ -124,6 +124,17 @@ public void shouldAllowSameStoreNameToBeUsedByMultipleNamedTopologies() {
         streams.start(asList(builder1.build(), builder2.build()));
     }
 
+    @Test
+    public void shouldAllowAddingAndRemovingNamedTopologyAndReturnBeforeCallingStart() throws Exception {
```

Comment:
This is a good test. But there is one more that I would like to see. It may have to be an integration test though. 

If we add two topologies that have overlapping source topics to an un-started client, will we see an error? Or will that error not be discovered until after the streams object is started? I think it will be the later, however I think we should document this behavior and have a test for it so that we know when it changes. I could see this becoming a source of much confusion later on.

Reply:
Discussed this offline, it's addressed by the new `shouldThrowTopologyExceptionWhenAddingNamedTopologyReadingFromSameInputTopicBeforeStart`  test